runtime/gc_modron_startup/arrayCopy.cpp

```diff
@@ -64,8 +64,9 @@ alwaysCallReferenceArrayCopyHelper(J9JavaVM *javaVM)
 static bool
 typeCheckArrayStore(J9VMThread *vmThread, J9Object *object, J9IndexableObject *arrayObj)
 {
+	J9Class *componentType = ((J9ArrayClass *)J9OBJECT_CLAZZ(vmThread, arrayObj))->componentType;
```

Comment:
For the case `object` and `arrayObj` are both NULL, this function returned true before this change. Now it will segfault.